src/fibex2som.rs

```diff
@@ -1149,6 +1167,68 @@ mod builder {
             _ => SOMTypeField::U32,
         }
     }
+
+    fn get_bitfield_type_items(type_decls: &[FibexTypeDeclaration]) -> Vec<SOMBitfieldMember> {
+        let mut items = Vec::new();
+
+        for type_decl in type_decls {
+            if type_decl.is_array() {
+                continue;
+            }
+
+            if let Some(bit_len) = type_decl.get_bit_length() {
+                if let Some(type_ref) = &type_decl.type_ref {
+                    let type_def = type_ref.borrow();
+
+                    let name = type_decl.name.clone();
+                    let description = type_def.name.clone();
+
+                    let datatype = match &type_def.datatype {
+                        FibexDatatype::Primitive(datatype) => datatype,
+                        FibexDatatype::Enum(datatype) => &datatype.primitive,
+                        _ => {
+                            continue;
+                        }
+                    };
+
+                    match datatype {
+                        FibexPrimitive::Uint8 => {
+                            items.push(SOMBitfieldMember::U8(SOMu8Bitfield::from(
+                                bit_len,
+                                SOMu8::empty().with_meta(SOMTypeMeta::from(name, description)),
+                            )));
+                        }
+                        FibexPrimitive::Uint16 => {
+                            items.push(SOMBitfieldMember::U16(SOMu16Bitfield::from(
+                                bit_len,
+                                SOMu16::empty(SOMEndian::Big)
+                                    .with_meta(SOMTypeMeta::from(name, description)),
+                            )));
+                        }
+                        FibexPrimitive::Uint32 => {
+                            items.push(SOMBitfieldMember::U32(SOMu32Bitfield::from(
+                                bit_len,
+                                SOMu32::empty(SOMEndian::Big)
+                                    .with_meta(SOMTypeMeta::from(name, description)),
+                            )));
+                        }
+                        FibexPrimitive::Uint64 => {
+                            items.push(SOMBitfieldMember::U64(SOMu64Bitfield::from(
+                                bit_len,
+                                SOMu64::empty(SOMEndian::Big)
+                                    .with_meta(SOMTypeMeta::from(name, description)),
+                            )));
+                        }
+                        _ => {
```

Comment:
same thing

Reply:
The number of unrelated variants is considerable big, therefore _ is applicable here.